traffic_ops/traffic_ops_golang/crconfig/handler.go

```diff
@@ -20,223 +20,146 @@ package crconfig
-
-		cdn, ok := params["cdn"]
-		if !ok {
-			handleErrs(http.StatusInternalServerError, errors.New("params missing CDN"))
-			return
-		}
-
-		ctx := r.Context()
-		user, err := auth.GetCurrentUser(ctx)
-		if err != nil {
-			handleErrs(http.StatusInternalServerError, err)
-			return
-		}
-
-		crConfig, err := Make(db.DB, cdn, user.UserName, r.Host, r.URL.Path, cfg.Version)
-		if err != nil {
-			handleErrs(http.StatusInternalServerError, err)
-			return
-		}
+func Handler(w http.ResponseWriter, r *http.Request) {
+	inf, userErr, sysErr, errCode := api.NewInfo(r, []string{"cdn"}, nil)
+	if userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, errCode, userErr, sysErr)
+		return
+	}
+	defer inf.Close()
 
-		resp := struct {
-			Response *tc.CRConfig `json:"response"`
-		}{crConfig}
-		respBts, err := json.Marshal(resp)
-		if err != nil {
-			handleErrs(http.StatusInternalServerError, err)
-			return
-		}
-		log.Infof("CRConfig time to generate: %+v\n", time.Since(start))
-		w.Header().Set("Content-Type", "application/json")
-		w.Write(respBts)
+	start := time.Now()
+	crConfig, err := Make(inf.Tx.Tx, inf.Params["cdn"], inf.User.UserName, r.Host, r.URL.Path, inf.Config.Version)
+	if err != nil {
+		api.HandleErr(w, r, http.StatusInternalServerError, nil, err)
+		return
 	}
+	log.Infof("CRConfig time to generate: %+v\n", time.Since(start))
+	api.WriteResp(w, r, crConfig)
 }
 
 // SnapshotGetHandler gets and serves the CRConfig from the snapshot table.
-func SnapshotGetHandler(db *sqlx.DB, cfg config.Config) http.HandlerFunc {
-	return func(w http.ResponseWriter, r *http.Request) {
-		handleErrs := tc.GetHandleErrorsFunc(w, r)
-		params, err := api.GetCombinedParams(r)
-		if err != nil {
-			handleErrs(http.StatusInternalServerError, err)
-			return
-		}
-		cdn, ok := params["cdn"]
-		if !ok {
-			handleErrs(http.StatusInternalServerError, errors.New("params missing CDN"))
-			return
-		}
+func SnapshotGetHandler(w http.ResponseWriter, r *http.Request) {
+	inf, userErr, sysErr, errCode := api.NewInfo(r, []string{"cdn"}, nil)
+	if userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, errCode, userErr, sysErr)
+		return
+	}
+	defer inf.Close()
 
-		snapshot, cdnExists, err := GetSnapshot(db.DB, cdn)
-		if err != nil {
-			handleErrs(http.StatusInternalServerError, errors.New("getting snapshot: "+err.Error()))
-			return
-		}
-		if !cdnExists {
-			handleErrs(http.StatusNotFound, errors.New("CDN not found"))
-			return
-		}
-		w.Header().Set("Content-Type", "application/json")
-		w.Write([]byte(`{"response":` + snapshot + `}`))
+	snapshot, cdnExists, err := GetSnapshot(inf.Tx.Tx, inf.Params["cdn"])
+	if err != nil {
+		api.HandleErr(w, r, http.StatusInternalServerError, nil, errors.New("getting snapshot: "+err.Error()))
+		return
 	}
+	if !cdnExists {
+		api.HandleErr(w, r, http.StatusNotFound, errors.New("CDN not found"), nil)
+		return
+	}
+	w.Header().Set(tc.ContentType, tc.ApplicationJson)
+	w.Write([]byte(`{"response":` + snapshot + `}`))
```

Comment:
It can't, because the snapshot in the database is the raw JSON, using the Go JSON tools to write it will write it as a string in quotes. 

I think it might work to make a `Response` object with a `json.RawMessage`, not positive. I thought about that, that might be slightly more typesafe; but, it's certainly faster to just write the string.